source/templates.tex

```diff
@@ -1612,13 +1613,68 @@
 \end{note}
 
 \pnum
-Two atomic constraints are
+Two atomic constraints, $e_1$ and $e_2$, are
 \indextext{identical!atomic constraints|see{atomic constraint, identical}}%
 \defnx{identical}{atomic constraint!identical}
 if they are formed from the same
 \grammarterm{expression}
-and the targets of the parameter mappings are equivalent
-according to the rules for expressions described in \ref{temp.over.link}.
+and if, given a hypothetical template $A$
+whose \grammarterm{template-parameter-list} consists of
+\grammarterm{template-parameter}s corresponding and equivalent\iref{temp.over.link} to
+those mapped by the parameter mappings of the expression,
+a \grammarterm{template-id} naming $A$
+whose \grammarterm{template-argument}s are
+the targets of the parameter mapping of $e_1$
+is the same\iref{temp.type} as
+a \grammarterm{template-id} naming $A$
+whose \grammarterm{template-argument}s are
+the targets of the parameter mapping of $e_2$.
+\begin{note}
+The comparison of parameter mappings of atomic constraints
+operates in a manner similar to that of declaration matching
+with alias template substitution\iref{temp.alias}.
+\begin{example}
+\begin{codeblock}
+template <unsigned N> constexpr bool Atomic = true;
+template <unsigned N> concept C = Atomic<N>;
+template <unsigned N> concept Add1 = C<N + 1>;
+template <unsigned N> concept AddOne = C<N + 1>;
+template <unsigned M> void f()
+  requires Add1<2 * M>;
+template <unsigned M> int f()
+  requires AddOne<2 * M> && true;
+
+int x = f<0>();     // OK, the atomic constraints from concept \tcode{C} in both \tcode{f}s are \tcode{Atomic<N>}
+                    // with mapping similar to $\tcode{N} \mapsto \tcode{2 * M + 1}$
+
+template <unsigned N> struct WrapN;
+template <unsigned N> using Add1Ty = WrapN<N + 1>;
+template <unsigned N> using AddOneTy = WrapN<N + 1>;
+template <unsigned M> void g(Add1Ty<2 * M> *);
+template <unsigned M> void g(AddOneTy<2 * M> *);
+
+void h() {
+  g<0>(nullptr);    // OK, there is only one \tcode{g}
+}
+\end{codeblock}
+\end{example}
+This similarity includes the situation where a program is ill-formed, no diagnostic required,
+when the meaning of the program depends on whether two constructs are equivalent,
+and they are functionally equivalent but not equivalent.
```

Comment:
What is "they"?  The "two constructs"?  How can they be "equivalent" and "not
equivalent" at the same time?

Reply:
That portion is a quote from [temp.over.link]/7. I am having a hard time finding another interpretation of "they" that would necessitate deviating from the formulation in [temp.over.link]. As for the latter question, they can (at the same time) be "functionally equivalent" and "not equivalent".

Reply:
This doesn't help me - I read "they" as the "two constraints", and this makes no sense.  Yes, entities can (at the same time) be "functionally equivalent" and "not equivalent". but here we're saying that they can be  "equivalent" and "not
equivalent" at the same time: "two constructs are equivalent,
and they are functionally equivalent but not equivalent."

Reply:
No, that's not what it says. It is talking about cases where the **program depends on whether they are equivalent**, but **they are not equivalent**. That's not "they are equivalent and not equivalent", it's "they need to be equivalent, but they are not".

Reply:
Ok, thanks to @jwakely, I finally get what this wording is trying to say, in which case the "and" should be changed to "but" at the very least, or preferably reworded so that it's not possible to parse it as I did.

Reply:
```suggestion
+but they are merely functionally equivalent and not equivalent.
```

Reply:
@hubert-reinterpretcast Much better, thank you!  Sadly, Richard has already merged this motion, so we'll have to create a separate PR for this and your suggestion below (for "one or more constraints").  Would you mind doing that since it's your wording?

Reply:
On reading this again, I am having trouble with the change I suggested above. I put in a more extensive change in #3807.